epp-client/src/connection/client.rs

```diff
@@ -187,10 +186,10 @@ impl EppClient {
 
         self.transact::<_, EppLogoutResponse>(&epp_logout).await
     }
-}
 
-impl Drop for EppClient {
-    fn drop(&mut self) {
-        let _ = block_on(self.logout());
+    pub async fn close(&mut self) -> Result<(), error::Error> {
```

Comment:
I think it might make sense to name this `shutdown` instead. I think the socket will be `close`d when the TcpStream goes out of scope. But it's still optimal for us to perform a clean `shutdown` explicitly.

https://beej.us/guide/bgnet/html/#close-and-shutdownget-outta-my-face

Reply:
You mean rename the `close()` in `EppConnection` or `EppClient` (or both)? The article is about shutting down socket connections, so I think you mean the one in `EppConnection`? This comment is against the function name in `EppClient`.

Reply:
Yeah, I got those mixed up.

Looking at it again, it might make sense to rename `close` to `shutdown` in `EppConnection` and then make the call to `shutdown` implicit as part of `logout`. So the caller just has to call `logout` and the function will know to call `shutdown`.

What do you think?

Reply:
> make the call to shutdown implicit as part of logout.

I thought it might be better to keep them separate in case a user needs to do a logout operation but would rather not renegotiate the socket connection if they still plan to log in and use the same connection.

But now that I checked back, the login call is coupled to the establishment of the connection too and it implies that an instance of `EppClient` is coupled to a set of login credentials, so yeah, would make sense to couple the `logout()` to the closing of the connection also. I've pushed the change for this.